src/utils/mod.rs

```diff
@@ -485,8 +486,16 @@ pub(crate) fn user_perms(cache: impl AsRef<Cache>, channel_id: ChannelId) -> Res
 /// ```
 #[inline]
 #[must_use]
-pub fn shard_id(guild_id: GuildId, shard_count: u32) -> u32 {
-    ((guild_id.get() >> 22) % (shard_count as u64)) as u32
+pub fn shard_id(guild_id: GuildId, shard_count: u16) -> u16 {
+    let shard_count = check_shard_total(shard_count);
+    ((guild_id.get() >> 22) % (shard_count.get() as u64)) as u16
+}
+
+pub(crate) fn check_shard_total(total_shards: u16) -> NonZeroU16 {
```

Comment:
nit: Do you think `shard_total_checked` is a better name?

Reply:
Not personally

Reply:
I'm pulling from methods like `u16::checked_add` for the naming here. When I first saw the method in review (before I saw the body), I thought it was somehow querying the value of the shard total, because of the method name wording (using the imperative). But tbh it's not a big deal.